### PR TITLE
[MISC] refactor stimuli mentioning sections in the events page

### PR DESCRIPTION
**Pull request by Remi-Gau**
PR related to #695 

- moves all sections about stimuli (whether about the stimuli folder or the database) together (including specific column names from the first table).

This is a draft looking for suggestions for improvements.

### Also included

- adding data types in the first table (leaving some `????` for values I am unsure about)

### Questions

- should the `identifier` and `database` columns be included in the tables of column names or were they there as just suggestions for the example?
- the `database` column is not described in the example JSON: should this be fixed?

### CODE REVIEW

File: src/04-modality-specific-files/05-task-events.md [hunk 1]
```diff
@@ -33,16 +33,15 @@ Each task events file REQUIRES a corresponding task imaging data file
...
+| response_time   | OPTIONAL              | [number][]    | Response time measured in seconds. A negative response time can be used to represent preemptive responses and "n/a" denotes a missed response.                                                                                                                                                                                                                     |
+| value           | OPTIONAL              | ????          | Marker value associated with the event (for example, the value of a TTL trigger that was recorded at the onset of the event).                                                                                                                                                                                                                                      |
```
Comment (Remi-Gau) [inline src/04-modality-specific-files/05-task-events.md:43]: Not sure what Datatype this is.

File: src/04-modality-specific-files/05-task-events.md [hunk 2]
```diff
@@ -33,16 +33,15 @@ Each task events file REQUIRES a corresponding task imaging data file
...
+| HED             | OPTIONAL              | ????          | Hierarchical Event Descriptor (HED) Tag. See [Appendix III](../99-appendices/03-hed.md) for details.                                                                                                                                                                                                                                                               |
```
Comment (Remi-Gau) [inline src/04-modality-specific-files/05-task-events.md:44]: Not sure if HED have a specific datatype that should be mentioned here.

tagging @VisLab

File: src/04-modality-specific-files/05-task-events.md (hunk 1 above)
Comment (sappelhoff) [reply src/04-modality-specific-files/05-task-events.md:43]: I have seen strings and numbers. For example in BrainVision files you can find a marker called `S  1` (string) that gets written when a TTL byte of `1` is received by the recording hardware. Then again some other hardware may simply encode that as `1` (number).

We could also keep it unspecified, or put something like `string, or number, or any other datatype`, I am not sure at all.

File: src/04-modality-specific-files/05-task-events.md (hunk 2 above)
Comment (sappelhoff) [reply src/04-modality-specific-files/05-task-events.md:44]: HED tags are usually strings, but I'll let Kay confirm just to be sure.

File: src/04-modality-specific-files/05-task-events.md [hunk 3]
```diff
@@ -1,48 +1,47 @@
...
+are supported (in contrast to "block" designs) - each "block of events" can be
+represented by an individual row in the `_events.tsv` file (with a long
+duration).
+
 Template:
 
 ```Text
 sub-<label>/[ses-<label>]
-    func/
+    <data_type>/
```
Comment (Remi-Gau) [inline src/04-modality-specific-files/05-task-events.md:17]: Events do not only concern `func` (at least mentioned in `beh`) so I suggest making that more general.

**On that note**

There are several bids-examples datasets for "eeg", "meg", "ieeg" that have `events.tsv`. But this seems to not be mentioned in the specifications that you can have those files there.

Should we at least mention in the `events` page what data type allow `events.tsv`?

And update the schema accordingly?

File: src/04-modality-specific-files/05-task-events.md (hunk 2 above)
Comment (Remi-Gau) [reply src/04-modality-specific-files/05-task-events.md:44]: @VisLab what would be the datatype for the HED column in an `event.tsv`

File: src/04-modality-specific-files/05-task-events.md (hunk 3 above)
Comment (Remi-Gau) [reply src/04-modality-specific-files/05-task-events.md:17]: Forget the part of the schema: I did not realize that the latest version in `master` has the events suffix for `eeg`, `meg`, `ieeg` and that this is rendered automatically in the specification upon build.

**Comment by VisLab** [comment]
The HED column type is a string.

File: src/04-modality-specific-files/05-task-events.md (hunk 1 above)
Comment (sappelhoff) [reply src/04-modality-specific-files/05-task-events.md:43]: looking at this again, let's say 
```
[string][] or [number][]
```

File: src/04-modality-specific-files/05-task-events.md (hunk 2 above)
Comment (sappelhoff) [reply src/04-modality-specific-files/05-task-events.md:44]: as reported here: https://github.com/bids-standard/bids-specification/pull/697#issuecomment-789128879, this should be a string

**Comment by Remi-Gau** [comment]
@effigies @sappelhoff 

Except for this failing link in a file I did not touch, this is good to go.

**Review by effigies — APPROVED** [review]

**Review by sappelhoff — APPROVED** [review]
Thanks @Remi-Gau

**Pull request merged**